docs/DeveloperGuide.md

```diff
@@ -1,38 +1,308 @@
+When the user deletes a module mapping, it will be added to the `ArrayDeque`.
+If the `ArrayDeque` already contains 5 module mappings, the last one (least recent) will be deleted, before the addition of a new module mapping.
+Do note that delete history information is not stored upon exiting the app (ie. it is not stored in User Storage).
+
+The following diagram illustrates the relationship between UserUniversityListManager and UserDeletedModules classes.
+
+![User Deleted Modules Class Diagram](./images/UserDeletedModules_Class.png)
+
+The following diagram illustrates the flow of the program, when a user deletes a module.
+
+![User Deleted Modules Sequence Diagram](./images/UserDeletedModules_Sequence.png)
+
+### 1.5 Ui
+
+The Ui class is the cornerstone of the Duke program to facilitate interaction with the user. It is used to scan and collect user input, print error messages to the user upon invalid input commands,
+and display the appropriate acknowledgements or required information based on the user's command.
+
+The following diagram illustrates the methods within the Ui class that can be invoked by the other classes in Duke for the purpose of user interaction.
+
+![Ui Class Diagram](./images/Ui_Class.png)
+
+### 1.6 Commands
+
+To interact with easySEP, users have to input commands specified with parameters to perform operations which will be passed to the CommandParser to generate a corresponding Command according to their input.
+The Command will then be executed to perform the operation on the other classes managing the timetables and databases. Error checking is handled to throw InvalidUserCommandException if the user's input does not match
+the specified Command format. To deal with parsing parameters, spaces in University names and Module codes are to be replaced with underscores.
+
+The following class diagram illustrates the relationship between Command class and its subclasses as well as other classes related to Commands.
+![Command Class Diagram](./images/Command_Class.png)
+
+#### 1.6.1 Create Command
+
+A Create command can be used to create a university list and its corresponding timetable.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a create command.
+
+![Create Command Sequence Diagram](./images/CreateCommand_Sequence.png)
+
+#### 1.6.2 Exit Command
+
+An Exit command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an exit command.
+
+![Exit Command Sequence Diagram](./images/ExitCommand_Sequence.png)
+
+#### 1.6.3 Help Command
+
+A Help command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a help command.
+
+![Help Command Sequence Diagram](./images/HelpCommand_Sequence.png)
+
```

Comment:
You are missing columns in your sequence diagram as UI, database, etc. are objects